examples/poisson_2d_mapping.py

```diff
@@ -22,15 +26,26 @@
 #==============================================================================
 class Laplacian:
 
-    def __init__( self, mapping ):
+#OLD def __init__( self, mapping ):
+#OLD     from psydac.mapping.analytical import AnalyticalMapping, IdentityMapping
+#OLD
+#OLD     assert isinstance( mapping, AnalyticalMapping )
+#OLD
+#OLD     sym = type(mapping).symbolic
+#OLD
+#OLD     self._eta        = sym.eta
+#OLD     self._metric     = sym.metric    .subs( mapping.params )
+#OLD     self._metric_det = sym.metric_det.subs( mapping.params )
+
+    def __init__(self, mapping):
 
-        assert isinstance( mapping, AnalyticalMapping )
+        assert isinstance(mapping, CallableMapping)
 
-        sym = type(mapping).symbolic
+        sym = mapping.symbolic_mapping
 
-        self._eta        = sym.eta
-        self._metric     = sym.metric    .subs( mapping.params )
-        self._metric_det = sym.metric_det.subs( mapping.params )
+        self._eta        = sym.logical_coordinates
+        self._metric     = sym.metric_expr    .subs(mapping.params)
+        self._metric_det = sym.metric_det_expr.subs(mapping.params)
```

Comment:
@saidctb Can a `CallableMapping` object have free parameters? The old `AnalyticalMapping` stored a dictionary of `(symbol, value)` pairs in `params`, but this looks empty now.

Reply:
yes, it can have free parameters, but was never tested,  we usually fix the free parameters when we create an instance of the Mapping class

Reply:
Do you mean [here](https://github.com/pyccel/sympde/blob/7e2a5622bb2eb3bf4a592db9113b1c9f9fdd4dcf/sympde/topology/callable_mapping.py#L20-L32)? I do not know how these lines could work:
```python
        constants    = mapping.constants
        params       = {a.name:a for a in constants}
        params.update( kwargs )
        for p in params.values():
            assert not isinstance(p, Symbol)


        if params:
            subs = {a:params[a.name] for a in constants}
            expressions = expressions.subs(subs)
            jac         = jac.subs(subs)
            inv_jac     = inv_jac.subs(subs)
            metric      = metric.subs(subs)
            metric_det  = metric_det.subs(subs)
```
What type of objects are in `mapping.constants`?

Reply:
I have just created issue #189 for discussing about this. I will now remove the commented lines and proceed with merging this small PR.

Reply:
okay, sounds good